src/resources/CaseAssistConfigs/CaseAssistConfigInterfaces.ts

```diff
@@ -0,0 +1,43 @@
+export enum DocumentSuggestionsStrategies {
+    ITD = 'ITD',
+}
+
+export enum TypingAidsStrategies {
+    StaticValues = 'StaticValues',
+    ValuesFromIndex = 'ValuesFromIndex',
+}
+
+export interface CaseAssistConfigListOptions {
+    page?: number;
+    pageSize?: number;
+}
+
+interface DocumentSuggestionsConfigurations {
+    ITD: {
+        pipeline?: string;
+        fieldsToFeed: string[];
+        filter?: string;
+        strategy: DocumentSuggestionsStrategies;
+    };
+}
+
+interface TypingAidsConfigurations {
+    StaticValues: {
+        values: string[];
+        strategy: TypingAidsStrategies;
+    };
+    ValuesFromIndex: {
+        filter: string;
+        field: string;
+        strategy: TypingAidsStrategies;
+    };
+}
+
+export interface CaseAssistConfigModel {
+    id: string;
+    name: string;
+    documentSuggestionConfiguration?: DocumentSuggestionsConfigurations[DocumentSuggestionsStrategies.ITD];
+    typingAidsConfiguration?:
+        | TypingAidsConfigurations[TypingAidsStrategies.StaticValues]
+        | TypingAidsConfigurations[TypingAidsStrategies.ValuesFromIndex];
+}
```

Comment:
These interfaces seem a bit overly complex to me, would something like the following work?

```ts
interface ITDConfigurationModel {
    pipeline?: string;
    fieldsToFeed: string[];
    filter?: string;
    strategy: DocumentSuggestionsStrategies;
}

interface StaticValuesConfigurationModel {
    values: string[];
    strategy: TypingAidsStrategies;
}

interface ValuesFromIndexConfigurationModel {
    filter: string;
    field: string;
    strategy: TypingAidsStrategies;
}


export interface CaseAssistConfigModel {
    id: string;
    name: string;
    documentSuggestionConfiguration?: ITDConfigurationModel;
    typingAidsConfiguration?: StaticValuesConfigurationModel | ValuesFromIndexConfigurationModel;
}
```

Reply:
I made them like that because we'll likely be adding new strategies to these lists in the future. I agree that right now it seems more complex than necessary, but I think it'll group things nicely if we have 4 or 5 strategies per category down the line. I'm thinking having a bunch of "loose" interfaces will get confusing. What do you think?

Reply:
If you want to group things together I would suggest going with an aggregating type such as

```ts
type TypingAidsConfigurations = StaticValuesConfigurationModel | ValuesFromIndexConfigurationModel | OtherTypingAids;
type DocumentSuggestionsConfigurations = ITDConfigurationModel | OtherDocumentSuggestionsTypes;
```

It scales much better in practice than having a fake object to define interfaces through a key:interface structure.

Reply:
I'll take your word for it